src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startWorkDate = LocalDate.parse(dateFrom, FORMATTER);
+        // converted from string dateFrom date to Date format
+        LocalDate finishWorkDate = LocalDate.parse(dateTo, FORMATTER);
+        // converted from string dateTo date to Date format
+        StringBuilder salaryInfo = new StringBuilder("Report for period "
+                    + dateFrom + " - "
+                    + dateTo);
+        for (String name : names) {
+            int salary = 0;
+            for (String allData : data) {
+                String[] dataInArray = allData.split(" ");
+                LocalDate certainDay = LocalDate.parse(dataInArray[0], FORMATTER);
+                // converted from string date in data to Date
+                int salaryPerDay = Integer.parseInt(dataInArray[2])
+                            * Integer.parseInt(dataInArray[3]);
```

Comment:
```suggestion
                int salaryPerDay = Integer.parseInt(dataInArray[SELARY_INDEX])
                            * Integer.parseInt(dataInArray[HOURS_INDEX]);
```
It is a good practice to make all values constants in your code